src/ProtocolGateway.Core/ErrorCode.cs

```diff
@@ -21,6 +21,7 @@ public enum ErrorCode
         ExactlyOnceQosNotSupported = 400011,
         UnknownPacketType = 400012,
         DuplicateConnectReceived = 400013,
+        ChannelClosed = 400014,
```

Comment:
I don't think `400` is appropriate here. How about https://httpstatuses.com/499?